libs/packages/components/src/lib/tree-table/tree-table.component.html

```diff
@@ -30,7 +30,7 @@
   let-parentRow="parentRow"
 >
   <tr
-    role="row"
+    role="treeitem"
```

Comment:
Using treeitem conveys the information that the row is part of a tree and provides the necessary context to understand the attributes aria-posinset, aria-setsize, and aria-expanded.

Reply:
before using treeitem I tested with row but it still gave the same error

Reply:
What was the error?

Per [ARIA 1.2 treeitem role](https://www.w3.org/TR/wai-aria-1.2/#treeitem)

> Authors MUST ensure [elements](https://www.w3.org/TR/wai-aria-1.2/#dfn-element) with [role](https://www.w3.org/TR/wai-aria-1.2/#dfn-role) treeitem are contained in, or [owned](https://www.w3.org/TR/wai-aria-1.2/#dfn-owned-element) by, an element with the role [group](https://www.w3.org/TR/wai-aria-1.2/#group) or [tree](https://www.w3.org/TR/wai-aria-1.2/#tree).

Also, there is a treegrid [example](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/examples/treegrid-1/) from the ARIA Authoring Practices Guide which I think pretty closely matches what we are trying to do. It is structured as a table with role=treegrid and within that table are multiple tr with role=row and attributes aria-posinset, aria-setsize, and aria-expanded all being used.

Reply:
this was the error while using role=row:
The aria-posinset attribute is not allowed on this TR because it does not have the correct role attribute
The aria-posinset and aria-setsize attribute pair is not allowed on this TR because it does not have the correct role attribute
The aria-setsize attribute is not allowed on this TR because it does not have the correct role attribute

changing the role to treeitem fixed it, but I am not sure why using role=row doest work even though it is documented that it should work fine.

Reply:
Did you get the error before switching the table row from grid to treegrid?

Reply:
i just tested it again with treegrid and row, it still shows the error.
treegrid and treeitem fixes the error.

Reply:
@davereed